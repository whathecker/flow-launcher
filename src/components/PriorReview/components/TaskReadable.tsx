import React from "react";
import { StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Container, Typography, Color } from "../../../styles";

type TaskReadableProps = {
  title: string;
};

const TaskReadable: React.FC<TaskReadableProps> = ({
  title,
}: TaskReadableProps) => {
  return (
    <View style={styles.taskWrapper}>
      <Text style={styles.text}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  taskWrapper: {
    ...Container.centerAligned,
    width: "85%",
    borderColor: Color.light.defaultBorder,
    borderRadius: 5,
    borderWidth: 1,
    paddingTop: 12,
    paddingBottom: 12,
    marginTop: 5,
    marginBottom: 5,
    marginLeft: "7.5%",
    marginRight: "7.5%",
  },
  text: {
    ...Typography.p,
    fontSize: 14,
  },
});

export default TaskReadable;
