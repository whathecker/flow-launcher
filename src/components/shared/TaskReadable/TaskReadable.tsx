import React from "react";
import { StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Container, Typography, Color } from "../../../styles";

type TaskReadableProps = {
  title: string;
};

const TaskReadable: React.FC<TaskReadableProps> = ({
  title,
}: TaskReadableProps) => {
  return (
    <View style={styles.taskWrapper}>
      <Text style={styles.text}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  taskWrapper: {
    ...Container.centerAligned,
    width: "100%",
    borderColor: Color.light.defaultBorder,
    borderRadius: 5,
    borderWidth: 1,
    paddingTop: 5,
    paddingBottom: 5,
    marginTop: 5,
    marginBottom: 5,
  },
  text: {
    ...Typography.p,
  },
});

export default TaskReadable;
