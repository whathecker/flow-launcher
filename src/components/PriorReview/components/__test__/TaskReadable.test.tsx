import * as React from "react";
import renderer from "react-test-renderer";
import TaskReadable from "../TaskReadable";

describe("Test TaskReadable component", () => {
  it("Should renders TaskReadable correctly", () => {
    const tree = renderer
      .create(
        <TaskReadable title="This should be title of empty prio bucket" />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
