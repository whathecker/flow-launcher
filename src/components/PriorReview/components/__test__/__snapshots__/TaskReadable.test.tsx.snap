// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TaskReadable component Should renders TaskReadable correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "marginLeft": "7.5%",
        "marginRight": "7.5%",
        "marginTop": 5,
        "paddingBottom": 8,
        "paddingTop": 8,
        "width": "85%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "Lexend-Regular",
          "fontSize": 14,
        },
      ]
    }
  >
    This should be title of empty prio bucket
  </Text>
</View>
`;
